my-app/src/components/buttons.tsx

```diff
@@ -1,10 +1,23 @@
 import * as React from "react";
 import "./buttons.css";
+import loadingIcon from "../images/ic_loading.svg";
 
-export class PrimaryButton extends React.Component{
+interface Props{
+    loading: boolean,
+    text: string,
+}
+
+export class PrimaryButton extends React.Component<Props,object>{
     render() {
+        let icon;
+        if(this.props.loading){
+            icon = <img src={loadingIcon} alt="loading" className="loadingIcon" />
+        }
         return (
-            <button className="primary" >Entrar</button>
+            <button className="primary">
+                {icon}
+                {this.props.text}
+            </button>
```

Comment:
Aqui você pode aplicar a técnica conhecida como [conditional rendering](https://reactjs.org/docs/conditional-rendering.html). Não sei se vc assistiu as aulas de Python, quando falamos de `and` (&& no javascript), mas ele vai avaliando valores da esquerda pra direita até achar um valor falsy ou até chegar no último:

```tsx
return (
  <button className="primary">
    {this.state.loading && <img src={loadingIcon} alt="loading" className="loadingIcon" />}
    {this.props.text}
  </button>
);
```